src/main/java/org/opentripplanner/api/common/RoutingResource.java

```diff
@@ -347,6 +351,48 @@
     @QueryParam("disableRemainingWeightHeuristic")
     protected Boolean disableRemainingWeightHeuristic;
 
+    /**
+     * Extra penalty added for flag stop boarding/alighting.
+     */
+    @QueryParam("flagStopExtraPenalty")
+    protected Integer flagStopExtraPenalty;
```

Comment:
All the penalties (including the existing penalties in OTP like `transferPenalty` and `otherThanPreferredRoutesPenalty`) are just costs that are added to state weights under different circumstances. As such, they don't exactly have units, but they are sort of "weighted seconds," e.g. how much time would you be willing to spend doing something else rather than board at a flag stop, although that something else may also be weighted if its walking or waiting. I can add some comments to that effect.

Re scoping: maybe. Currently there isn't any similar scoping among the OTP parameters, so we would be introducing a convention. Perhaps all GTFS-flex parameters could be prefixed with "flex_"? Or we could create a new object for GtfsFlexOptions with all the parameters for flex included? Perhaps the second option scales better as more features get added to OTP...

Reply:
I am not sure how the second option would map to HTTP GET parameters, but the prefix would work. I know you are just following the existing pattern - some of the bike properties have the exact same issue. What is your thought about this @abyrd?

Reply:
The issues @t2gran raised are also present in existing API parameters. But an API clean-up is on the road map so we should start establishing principles. Considering the large amount of parameters, we should add scoping in the form of prefixes. 

The idea of adding a new GTFSFlexOptions object makes sense within OTP, but it's not clear how that would map to HTTP API parameters.

I am in favor of using prefixes and adding lines to the Javadoc ("applies to flex routing only").

As @sdjacobs says all penalties and costs (hence the whole generalized cost of a path) are implicitly expressed in in-vehicle seconds, since that's one of the few things that doesn't usually a cost multiplier applied to it. That should be clarified in the documentation or API javadoc but it's a pervasive problem, not a problem with this PR.

Reply:
There is an additional discussion to be had though about what is a request-scoped HTTP API parameter, and what is a system-wide parameter set in config JSON. That issue has come up in other comments in @t2gran 's review, and I'll comment on it in those places.

Reply:
I was talking to @t2gran about this today - we noticed there are several new query parameters here that may not really be request-scoped HTTP API parameter, but rather system-wide parameters set in config JSON. 

@sdjacobs note that it's possible to have config values in RoutingRequest that are not exposed in the HTTP API. Are all of the new values really intended to be set freely on a per-request basis?

Reply:
I will go through and add a "flex" prefix and notes to the Javadoc. (I suppose this is creating a new convention - is "flex" in camelcase ok, e.g. "flexFlagStopExtraPenalty"? or perhaps "flex_flagStopExtraPenalty"?)

The only API parameters we are currently setting per-request are `useReservationServices` and `useEligibilityServices`. I think `ignoreDrtAdvanceBookMin` should also be settable, because different UIs may make different decisions on whether its appropriate to show a call-and-ride trip today even if you can't actually take it if you haven't reserved it (and this could be a user preference.) But in any case, I lean towards allowing more internal settings to be overrideable via the API, to make it easier to experiment and find appropriate values, or for advanced users. Thoughts? If folks disagree it's easy enough for me to pull those lines from `RoutingResource`.